packages/framework/presence/src/notificationsManager.ts

```diff
@@ -0,0 +1,140 @@
+> {
+	/**
+	 * Register a notification listener.
+	 * @param notificationName - the name of the notification
+	 * @param listener - the handler to run when the notification is received from other client
+	 * @returns a function which will deregister the listener when run. This function
+	 * has undefined behavior if called more than once.
+	 */
+	on<K extends keyof InternalUtilityTypes.NotificationEvents<E>>(
+		notificationName: K,
+		listener: (
+			sender: ConnectedClientId,
+			...args: InternalUtilityTypes.JsonDeserializedParameters<E[K]>
+		) => void,
+	): () => void;
+}
+
+/**
+ * Record of notification subscriptions.
+ *
+ * @sealed
+ * @alpha
+ */
+export type NotificationSubscriptions<E extends InternalUtilityTypes.NotificationEvents<E>> = {
+	[K in string & keyof InternalUtilityTypes.NotificationEvents<E>]: (
+		sender: ConnectedClientId,
+		...args: InternalUtilityTypes.JsonSerializableParameters<E[K]>
+	) => void;
+};
+
+/**
+ * Interface for a notification emitter that can send typed notification to other clients.
+ *
+ * @sealed
+ * @alpha
+ */
+export interface NotificationEmitter<E extends InternalUtilityTypes.NotificationEvents<E>> {
+	/**
+	 * Emits a notification with the specified name and arguments, notifying all clients.
+	 * @param notificationName - the name of the notification to fire
+	 * @param args - the arguments sent with the notification
+	 */
+	broadcast<K extends string & keyof InternalUtilityTypes.NotificationEvents<E>>(
+		notificationName: K,
+		...args: Parameters<E[K]>
+	): void;
+
+	/**
+	 * Emits a notification with the specified name and arguments, notifying a single client.
+	 * @param notificationName - the name of the notification to fire
+	 * @param targetClientId - the single client to notify
+	 * @param args - the arguments sent with the notification
+	 */
+	unicast<K extends string & keyof InternalUtilityTypes.NotificationEvents<E>>(
+		notificationName: K,
+		targetClientId: ConnectedClientId,
+		...args: Parameters<E[K]>
+	): void;
+}
+
+/**
+ * Value manager that provides notifications from this client to others and subscription
+ * to their notifications.
+ *
+ * @remarks Create using {@link Latest} registered to {@link PresenceStates}.
+ *
+ * @sealed
+ * @alpha
+ */
+export interface NotificationsManager<T extends InternalUtilityTypes.NotificationEvents<T>> {
+	/**
+	 * Events for Notifications manager.
+	 */
+	readonly events: ISubscribable<NotificationsManagerEvents>;
+
+	/**
+	 * Send notifications to other clients.
+	 */
+	readonly emit: NotificationEmitter<T>;
+
+	/**
+	 * Provides subscription to notifications from other clients.
+	 */
+	readonly notifications: NotificationSubscribable<T>;
+}
+
+/**
+ * Factory for creating a {@link NotificationsManager}.
+ *
+ * @alpha
+ */
+export function Notifications<
+	T extends InternalUtilityTypes.NotificationEvents<T>,
+	Key extends string,
+>(
+	initialSubscriptions: NotificationSubscriptions<T>,
```

Comment:
They could go one by one, but I think there goodness. Initial subscriptions provides opportunity for complete schema of notifications. Customers could build that themselves but if it valuable (and I am judging that it is), then we can provide. A negative of one-by-one is that there could be collision.
Note that this isn't optional, but it may be empty. Optional just makes it harder to derive the schema and requiring `{}` isn't a burden.

Reply:
It is also worth noting that the a notifications workspace has division of Notifications. The workspace is a record of Notifications. Notifications then has named-typed "events". There is nothing currently that notes when a "division" hasn't been instantiated to listen. (Same is true for States.)